gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -166,7 +166,28 @@ impl GatewayLightning for ClnRpcService {
         &self,
         _request: tonic::Request<GetPubKeyRequest>,
     ) -> Result<tonic::Response<GetPubKeyResponse>, Status> {
-        unimplemented!()
+        self.client
+            .lock()
+            .await
+            .call(cln_rpc::Request::Getinfo(
+                model::requests::GetinfoRequest {},
+            ))
+            .await
+            .map(|response| {
+                let pub_key = match response {
+                    cln_rpc::Response::Getinfo(model::responses::GetinfoResponse {
+                        id, ..
+                    }) => id,
+                    _ => panic!("Unexpected response from cln_rpc"),
```

Comment:
I'd prefer to avoid new `panic`s at this point when it isn't a logic error on our side. This one is borderline ok because we might not want to run the GW anyway if the CLN API schema changed. thoughts @dpc?

Reply:
Seems like this error  mean that our implementation is inherently incorrect? (wrong schema w.r.t. to node we're paired with? In which case this is excusable (or maybe even a good idea).

So... put a longer panic message and good enough?